plugin/src/rules/prefer_promise_reject_errors.rs

```diff
@@ -0,0 +1,358 @@
+                    },
+                    {
+                        code => "new Promise(function({}, reject) { reject(5) })",
+                        errors => errors,
+                    },
+                    {
+                        code => "new Promise(({}, reject) => reject(5))",
+                        errors => errors,
+                    },
+                    {
+                        code => "new Promise((resolve, reject, somethingElse = reject(5)) => {})",
+                        errors => errors,
+                    },
+
+                    // Optional chaining
+                    {
+                        code => "Promise.reject?.(5)",
+                        errors => errors,
+                    },
+                    {
+                        code => "Promise?.reject(5)",
+                        errors => errors,
+                    },
+                    {
+                        code => "Promise?.reject?.(5)",
+                        errors => errors,
+                    },
+                    {
+                        code => "(Promise?.reject)(5)",
+                        errors => errors,
+                    },
+                    {
+                        code => "(Promise?.reject)?.(5)",
+                        errors => errors,
+                    },
+
+                    // Assignments with mathematical operators will either evaluate to a primitive value or throw a TypeError
+                    {
+                        code => "Promise.reject(foo += new Error())",
+                        errors => errors,
+                    },
+                    {
+                        code => "Promise.reject(foo -= new Error())",
+                        errors => errors,
+                    },
+                    {
+                        code => "Promise.reject(foo **= new Error())",
+                        errors => errors,
+                    },
+                    {
+                        code => "Promise.reject(foo <<= new Error())",
+                        errors => errors,
+                    },
+                    {
+                        code => "Promise.reject(foo |= new Error())",
+                        errors => errors,
+                    },
+                    {
+                        code => "Promise.reject(foo &= new Error())",
+                        errors => errors,
+                    },
+
+                    // evaluates either to a falsy value of `foo` (which, then, cannot be an Error object), or to `5`
+                    {
+                        code => "Promise.reject(foo && 5)",
+                        errors => errors,
+                    },
+                    {
+                        code => "Promise.reject(foo &&= 5)",
+                        errors => errors,
+                    },
+                ]
+            },
+            get_instance_provider_factory(),
+            json_object!({
+                "ecma_version": 2022,
```

Comment:
Without this one of the test cases was failing because I guess it wasn't treating the `const reject = ...` "overriding" declaration the way you'd expect (ie in the ES6 way)